### PR TITLE
Add docs for getNFTSales

### PR DESCRIPTION
**Pull request by alex-miao**

### CODE REVIEW

File: nft/nfts.yaml [hunk 1]
```diff
@@ -649,6 +649,105 @@ paths:
...
                     description: 'Whether the given wallet owns any token in the given NFT collection.'
       operationId: isHolderOfCollection
+  '/{apiKey}/getNFTSales':
+    get:
+      summary: getNFTSales
+      description: Gets NFT sales that have happened through on-chain marketplaces
+      tags: ['NFT API Endpoints']
+      parameters:
+        - $ref: '#/components/schemas/tokenId'
+        - $ref: nfts_body.yaml#/startBlock
```
Comment (elanh) [inline nft/nfts.yaml:659]: nfts_body.yaml no longer exists, you can reference things in the components/schemas section below!

File: nft/nfts.yaml [hunk 2]
```diff
@@ -649,6 +649,105 @@ paths:
...
+          description: String - The block number to start fetching NFT sales data from. Allowed values are decimal integers and "latest". Defaults to "latest".
+        - $ref: nfts_body.yaml#/startLogIndex
+          description: Integer - The log index within the startBlock to start fetching NFT sales data from.
```
Comment (elanh) [inline nft/nfts.yaml:662]: let's add all descriptions into the components section below in case they need to be reused

**Review by elanh — CHANGES_REQUESTED** [review]
slight changes with these

File: nft/nfts.yaml [hunk 3]
```diff
@@ -188,7 +188,7 @@ paths:
                         type: string
                         description: 'String - Total number of NFTs in a given NFT collection.'
                       tokenType:
-                        $ref: '#/components/schemas/tokenType_response'   
+                        $ref: '#/components/schemas/tokenType_response'
```
Comment (elanh) [inline nft/nfts.yaml:191]: @alex-miao did u manually remove this or do u have some sort of linter that does this for you? would love to set it up lol

File: nft/nfts.yaml [hunk 4]
```diff
@@ -649,6 +649,99 @@ paths:
...
+          description: String - The contract address of a NFT collection to filter sales by.
+          required: false
+        - $ref: '#/components/schemas/tokenId'
+          description: String - The token id of an NFT within the collection specified by contractAddress to filter sales by.
+          required: false
+        - $ref: '#/components/schemas/buyerAddress'
+        - $ref: '#/components/schemas/sellerAddress'
+        - $ref: '#/components/schemas/buyerIsMaker'
+        - $ref: '#/components/schemas/limit'
+          description: Integer - The maximum number of NFT sales return. Defaults to 100.
```
Comment (elanh) [inline nft/nfts.yaml:673]: missing "to" -->  "The maximum number of NFT sales **to** return"

File: nft/nfts.yaml [hunk 5]
```diff
@@ -773,6 +866,67 @@ components:
...
+    startLogIndex:
+      name: startLogIndex
+      description: Integer - The log index within the startBlock to start fetching NFT sales data from.
+      in: query
+      schema:
+        type: integer
+      required: false
+    startBundleIndex:
+      name: startBundleIndex
+      description: Integer - The index of an NFT within a sale bundle to start fetching NFT sales data from.
```
Comment (elanh) [inline nft/nfts.yaml:886]: i probably am missing context here but is "sale bundle" a common this or something we decided?

**Review by elanh — APPROVED** [review]

File: nft/nfts.yaml (hunk 3 above)
Comment (alex-miao) [reply nft/nfts.yaml:191]: Think it was a linter on VS Code, not certain which one but will send you the list of extensions I have.

File: nft/nfts.yaml (hunk 5 above)
Comment (alex-miao) [reply nft/nfts.yaml:886]: This is terminology from Seaport but the interface can also be applied to other marketplaces that support batched/bundled NFT sales.

**Pull request merged**